lib/report.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Report do
+  @doc """
+  Exectutes a report against a given repo, returning the results or an error
+  """
+  @type t :: module()
+
+  @callback run() :: {:ok, [map()]} | :error
+  @callback short_name() :: String.t()
+  @callback description() :: String.t()
+
+  @spec report_to_csv(t()) :: {:ok, String.t()} | :error
+  def report_to_csv(report) do
```

Comment:
small point here, I would suggest the function just be called `to_csv`. The `report` in `report_to_csv` is redundant with the module name.

Reply:
Good point, I'll make the change.